cdap-security/src/main/java/co/cask/cdap/security/authz/AuthorizationHandler.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authz;
+
+import co.cask.cdap.proto.id.EntityId;
+
+import java.util.Set;
+
+/**
+ * Authorization plugin interface.
+ */
+public interface AuthorizationHandler {
+  /**
+   * Checks if a user is allowed to perform a set of actions on an entity.
+   *
+   * @param entity the entity
+   * @param user the user
+   * @param actions the actions (wildcard: '*')
+   * @return true if the user is allowed to perform all of the actions on the entity
+   */
+  boolean authorized(EntityId entity, String user, Set<String> actions);
+
+  /**
+   * Grants a user permission to perform a set of actions on an entity.
+   *
+   * @param entity the entity
+   * @param user the user
+   * @param actions the actions
+   */
+  void grant(EntityId entity, String user, Set<String> actions);
```

Comment:
But in practice, what are the set of possible actions? Is it all up to user define (hence string) or always be some defined type (e.g. READ/INSERT/UPDATE/DELETE... ).. if it's the latter case, then using enum make sense.